record-core/src/androidMain/kotlin/dev/theolm/record/RecordCore.android.kt

```diff
@@ -4,60 +4,103 @@ package dev.theolm.record
 
 import android.Manifest.permission.RECORD_AUDIO
 import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.media.AudioRecord
 import android.media.MediaRecorder
 import android.os.Build
 import androidx.core.content.ContextCompat
-import dev.theolm.record.config.AudioEncoder
 import dev.theolm.record.config.OutputFormat
-import dev.theolm.record.config.OutputLocation
 import dev.theolm.record.config.RecordConfig
 import dev.theolm.record.error.NoOutputFileException
 import dev.theolm.record.error.PermissionMissingException
 import dev.theolm.record.error.RecordFailException
+import java.io.FileOutputStream
 
 @Suppress("TooGenericExceptionCaught", "SwallowedException")
 internal actual object RecordCore {
+    private var audioRecord: AudioRecord? = null
+    private var recordingThread: Thread? = null
     private var recorder: MediaRecorder? = null
     private var output: String? = null
-    private var recordingState : RecordingState = RecordingState.IDLE
+    private var myRecordingState : RecordingState = RecordingState.IDLE
```

Comment:
any reason for changing the state name?
I prefer using the old one.

Reply:
I remember the name conflicted with another variable somewhere.